Controller/RegistrationController.php

```diff
@@ -70,9 +70,16 @@ public function registerAction(Request $request)
             return $response;
         }
 
-        return $this->render('FOSUserBundle:Registration:register.html.twig', array(
-            'form' => $form->createView(),
-        ));
+        $event = new FormEvent($form, $request);
+        $dispatcher->dispatch(FOSUserEvents::REGISTRATION_FAILURE, $event);
```

Comment:
This event will be triggered even if the form has not been submitted.

How about:

```php
if ($form->isSubmitted()) {
    if ($form->isValid()) {
        // Update user
    } else {
        $event = new FormEvent($form, $request);
        $dispatcher->dispatch(FOSUserEvents::REGISTRATION_FAILURE, $event);
    }
}
```